File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        opening = ['{','[','(']
        closing = [')','}',']']
        stack = []
        for letter in s:
            if letter in opening:
                stack.append(letter)
            else:
                if len(stack) ==0:
                    return False
                elif letter == '}':
                    if stack[-1] != '{':
                        return False
                    else:
                        stack.pop()
                elif letter == ')':
                    if stack[-1] != '(':
                        return False
                    else:
                        stack.pop()
                elif letter == ']':
                    if stack[-1] != '[':
                        return False
                    else:
                        stack.pop()
        if len(stack) == 0:
            return True
        else:
            return False

            



        

File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        vector <int> nums ;
        for(int i = 0;i<s.length();i++) {
            if (s[i] == '{' || s[i]=='[' || s[i]=='(') {
                nums.push_back(s[i]);
            }
            else if (s[i] == '}' && nums.size() >0) {
                if (nums[nums.size()-1] == '{') {
                    nums.pop_back();
                } else {
                    return false;
                }
            } 
            else if (s[i] == ']' && nums.size() >0) {
                if (nums[nums.size()-1] == '[') {
                    nums.pop_back();
                } else {
                    return false;
                }
            }
            else if (s[i] == ')' && nums.size() >0) {
                if (nums[nums.size()-1] == '(') {
                    nums.pop_back();
                } else {
                    return false;
                }
            } else {
                return false;
            }
        }
        if (nums.size() == 0) {
        
        return true;} else {return false;}
    }
};